domain/model/state/types.go

```diff
@@ -205,3 +205,34 @@ type dbController struct {
 type dbUserName struct {
 	Name string `db:"name"`
 }
+
+// dbModelUserInfo represents information about a user on a particular model.
+type dbModelUserInfo struct {
+	// Name is the username of the user.
+	Name string `db:"name"`
+
+	// DisplayName is a user-friendly name represent the user as.
+	DisplayName string `db:"display_name"`
+
+	// LastModelLogin is the last time the user logged in to the model.
+	LastModelLogin time.Time `db:"time"`
+
+	// AccessType is the access level the user has on the model.
+	AccessType string `db:"access_type"`
+}
+
+// toModelUserInfo converts the result from the DB to the type
+// access.ModelUserInfo.
+func (info *dbModelUserInfo) toModelUserInfo() (coremodel.ModelUserInfo, error) {
```

Comment:
**thought:** I don't like this kind of transformation function, it is so easy to silently break it by adding new fields to dbModelUserInfo or ModelUserinfo. However I understand the intent and I have no idea to avoid this pitfall.

By "silently break", I mean "perform changes who cause bad behavior without breaking TU".